external/mmdetection/detection_tasks/apis/detection/openvino_task.py

```diff
@@ -456,17 +457,48 @@ def infer(
                 )
 
             if add_saliency_map and saliency_map is not None:
-                saliency_map = get_actmap(
-                    saliency_map, (dataset_item.width, dataset_item.height)
-                )
-                saliency_map_media = ResultMediaEntity(
-                    name="Saliency Map",
-                    type="saliency_map",
-                    annotation_scene=dataset_item.annotation_scene,
-                    numpy=saliency_map,
-                    roi=dataset_item.roi,
-                )
-                dataset_item.append_metadata_item(saliency_map_media, model=self.model)
+                if saliency_map.ndim == 2:
+                    # Single saliency map per image, support e.g. EigenCAM use case
+                    actmap = get_actmap(
+                        saliency_map, (dataset_item.width, dataset_item.height)
+                    )
+                    saliency_media = ResultMediaEntity(
+                        name="Saliency Map",
+                        type="saliency_map",
+                        annotation_scene=dataset_item.annotation_scene,
+                        numpy=actmap,
+                        roi=dataset_item.roi
+                    )
+                    dataset_item.append_metadata_item(saliency_media, model=self.model)
+                elif saliency_map.ndim == 3:
+                    # Multiple saliency maps per image (class-wise saliency map)
+                    predicted_class_set = set()
+                    for bbox in predicted_scene.annotations:
+                        predicted_class_set.add(bbox.get_labels()[0].name)
+
+                    labels = self.task_environment.get_labels()
+                    num_saliency_maps = saliency_map.shape[0]
+                    if num_saliency_maps == len(labels) + 1:
+                        # Include the background as the last category
+                        labels.append(LabelEntity('background', Domain.DETECTION))
+                    for class_id, class_wise_saliency_map in enumerate(saliency_map):
+                        class_name_str = labels[class_id].name
+                        if class_name_str in predicted_class_set:
+                            # TODO (negvet): Support more advanced use case,
+                            #  when all/configurable set of saliency maps is returned
```

Comment:
I was requested to make only predicted classes to be explained for Geti (at least for Geti1.1).
This is due to possible memory overhead issues in the many classes use case.
Applies only for OV tasks.

Also update cls in this PR (I know, I better do it in a separate PR, sorry - I am in a bit hurry)

@goodsong81 @sungmanc @wonjuleee please take a look

Reply:
Okay, it looks like reasonable.

Reply:
LGTM